contracts/ERC721.sol

```diff
@@ -1233,4 +1235,15 @@ contract TradeTrustERC721 is ERC721MintableFull, IERC721Receiver {
   {
     return this.onERC721Received.selector;
   }
+
+  function destroyToken(uint256 _tokenId) public onlyMinter {
+    require(ownerOf(_tokenId) == address(this), "Cannot destroy token: Token not owned by token registry");
+    _burn(_tokenId);
+  }
+
+  function sendToken(address to, uint256 _tokenId) public onlyMinter {
+    require(ownerOf(_tokenId) == address(this), "Cannot send token: Token not owned by token registry");
+    _safeTransferFrom(ownerOf(_tokenId), to, _tokenId, "");
+  }
+
 }
```

Comment:
I think you will need to be able to receive token in this contract, so that both safeMint and safeTransfer can send token here.

Reply:
Hmm I believe that is already done through the `onERC721Received` function on line 1232 which is called when you do a safeMint/safeTransfer

Reply:
Ah i didnt see that in the code diff. Can ignore this comment but there another concerns.
You want to emit an event when a token is received so that it can be indexed and search via the web app if we want to detect surrender action.